proxy/src/error.rs

```diff
@@ -29,7 +29,7 @@ pub trait UserFacingError: fmt::Display {
     }
 }
 
-#[derive(Clone)]
+#[derive(Copy, Clone, Debug)]
 pub enum ErrorKind {
```

Comment:
On the second thoughts, I think, it's rather `reason`, than `kind`, but I am not sure. WDYT?

Reply:
I think both options seem fine